File: benchs/concat_compare.f
# List-heavy benchmark for cross-language comparison.

row(n): [1 | 0 <= i < n].

concat_rows(0, _, acc): acc.
concat_rows(k, n, acc): concat_rows(k - 1, n, [acc] ++ [row(n)]).

main():
    res <- concat_rows(3000, 8, [])
    say(len(res))
    1.

File: benchmarks/bench/concat_compare.py
#!/usr/bin/env python3


def row(n: int):
    return [1 for _ in range(n)]


def concat_rows(k: int, n: int, acc):
    for _ in range(k):
        acc = acc + row(n)
    return acc


def main() -> int:
    res = concat_rows(3000, 8, [])
    print(len(res))
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
